#!/usr/bin/env python
PACKAGE = "label_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("scaled_height", int_t, 0, "An Integer parameter", 400,  0, 1125)
gen.add("v_thresh", int_t, 0, "Value channel threshold", 90, 0, 255)
gen.add("s_thresh", int_t, 0, "Saturation channel threshold", 130, 0, 255)
gen.add("median_ksize_ratio", double_t, 0,
        "Median kernel size relative to scaled_height", 0.01, 0, 0.5)
gen.add("use_rectangles", bool_t, 0,
        "Use rectangular vs elliptical morphological structuring elements",
        False)
gen.add("tophat_ratio", double_t, 0,
        "Tophat structuring element width and height relative to scaled_height",
        0.125, 0, 1)
gen.add("close_ratio", double_t, 0,
        "Close structuring element width and height relative to scaled_height",
        0.04, 0, 1)
gen.add("open_width_ratio", double_t, 0,
        "Open structuring element width relative to scaled_height",
        0.075, 0, 1)
gen.add("open_height_ratio", double_t, 0,
        "Open structuring element height relative to scaled_height",
        0.060, 0, 1)

exit(gen.generate(PACKAGE, "label_detector", "Detector"))
